lib/src/OS/Windows/SharedMemory.cc

```diff
@@ -0,0 +1,69 @@
+#include <windows.h>
+#include <string>
+#include <stdexcept>
+
+#include "Error.h"
+#include "OS/SharedMemory.h"
+
+namespace kickcat
+{
+    #define  THROW_LAST_ERROR(msg) (throw std::system_error(static_cast<int>(GetLastError()), std::system_category(),  LOCATION(": " msg)))
+
+    SharedMemory::SharedMemory()
+        : fd_{nullptr}
+    {
+    }
+
+    SharedMemory::~SharedMemory()
+    {
+        if (address_ != nullptr)
+        {
+            if (not UnmapViewOfFile(address_))
+            {
+                THROW_LAST_ERROR("UnmapViewOfFile() failed");
```

Comment:
do you really want to throw in destruction? Maybe just logging might be better, functionally it has no impact just the clean up failed we do not want the application to throw for this non?